### DISCUSSION TITLE
Need help in setting up Stockfish in a 512 core server

### DISCUSSION THREAD

**Discussion by Thalaivachess** [official-stockfish/Stockfish#4346] [Q&A]
Need help in configuring Stockfish in a 512 core machine

**Comment by vondele** [comment]
In principle nothing particular needed, except for setting the number of Threads and Hash to a large value. 
Do you have some information on the particular hardware you use, that might be useful.

**Comment by Thalaivachess** [comment]
We are getting access to a powerful machine from AWS to help in our
preparations.
We want to use the hardware and SF in the best possible manner.

Processor Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz   2.90 GHz (we are
currently using 8 server nodes of 64 core each )
Installed RAM 32.0 GB (31.5 GB usable)
System type 64-bit operating system, x64-based processor

**Comment by Sopel97** [comment]
At this point I'd either invest in a dedicated dual EPYC machine with at least 512GB of RAM, or look at chessify. I skimmed through AWS and they don't seem to have any reasonable configuration for chess analysis (there are high vcore machines but they are expensive and have excessive RAM). The multi-node setup with those intel cpus might be expensive and painful to use compared to alternatives, and the amount of RAM there will cripple analysis significantly.

**Comment by Ipmanchess** [comment]
If i understand well ,then they need Stockfish Cluster version..

**Reply by Thalaivachess** [reply]
Can I get more information on this, please? We have a technical team willing to help us but they are mostly software professionals without any knowledge of chess/stockfish etc. If there is a document I can share with them to get the best possible outcome, it will be greatly helpful. This is for a top Chess Grandmaster with high ambitions. 
Currently, we have access to this machine: 
Processor Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz   2.90 GHz (we are
 using 8 server nodes of 64 core each = 512 cores)
Installed RAM 32.0 GB (31.5 GB usable)
System type 64-bit operating system, x64-based processor

**Reply by Sopel97** [reply]
good start https://github.com/official-stockfish/Stockfish/wiki/Advanced-topics#cluster-version

**Comment by Ipmanchess** [comment]
And if you get Stockfish Cluster working on that server ,then i hope you are willing to run this bench : [https://ipmanchess.yolasite.com/amd--intel-chess-bench-stockfish.php](url) ,then you directly have a idea how fast that system is!

PS> Looks like my link doesn't open ,just search for ipmanchess or copy link into your browser.

**Reply by Thalaivachess** [reply]
Will convey to my team, thanks!

**Comment by mstembera** [comment]
Before we start recommending SF cluster version has there ever been any testing done characterizing cluster versions performance/efficiency?  I don't mean nps but actual play testing measuring elo.  Obviously having cores spread across multiple machines is much less efficient than having them all on one machine because each machine has a separate hash which can only be shared over the network which is super slow compared to being local.
Edit:  A good basic test would be to play regular SF on one 64 core AWS machine vs cluster SF on two 64 core each AWS machines and measure how much elo that brings.

@Thalaivachess Having a real 512 core machine would be much better than 8 separate 64 core machines.  Even a lower core single machine would likely be much better but we need testing to determine the exact trade off.

**Reply by Sopel97** [reply]
yea, I'm thinking along the same line. Also the RAM amount is really low in this cluster setting.

**Reply by Vizvezdenec** [reply]
Vondele did some testing and clusterfish definitely plays stronger than 1 issue of the same machine. Efficiency isn't that great but it's better than nothing.
Of course if you can get dual AMD EPYC instead it will do much better.

**Reply by Thalaivachess** [reply]
Thanks for the suggestions. With the current setup, how much RAM would be ideal pls? And the ideal thread number as well please.

**Reply by vondele** [reply]
Each of these servers could run an analysis independently, probably using 
setoption name Threads value 63
setoption name Hash value 30000
i.e. essentially using all resources. (Check the system has still a bit of free memory when the analysis is running).

The cluster version of SF could be used to combine all servers for a single analysis. However, this is technically a bit more difficult (i.e. needs a properly working MPI setup, and effectively building a real cluster in the cloud. Can be done, but non-trivial). Unless you're very determined to get the cluster  to work, I would first consider using to different server instances (i.e. dual epyc with more cores), before trying to setup the cluster version.
